<a href="https://colab.research.google.com/github/facundoveron/colab_datasciense_1_coder/blob/main/airbnb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ingesta de Datos y Primer Vistazo

## Instalacion de librerias

In [17]:
!pip install --upgrade numpy \
  pandas

## Importar librerias Pandas y Numpy

In [18]:
import numpy as np
import pandas as pd

## Cargamos dataset

In [19]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/train.csv')

## Mostramos los primero 5 registros de nuestro dataset

In [20]:
df.head(n=5)

,id,name,description,neighborhood_overview,host_id,host_name,host_since,host_location,host_about,host_response_time,...,last_review,review_scores_rating,review_scores_accuracy,review_scores_cleanliness,review_scores_checkin,review_scores_communication,review_scores_location,review_scores_value,reviews_per_month,city
0,1234376912864523021,Spacious Garden Apt in Center Square,Nestled in the vibrant heart of the historic C...,NaN,564967431,Stephanie,2024-03-02,NaN,NaN,within an hour,...,2024-10-28,5.00,5.00,5.00,5.00,5.00,5.00,5.00,0.44,Albany
1,887167811237271928,The Comfort II,Your family will be close to everything when y...,NaN,459472925,Gabriel,2022-05-16,"Albany, NY",NaN,within a few hours,...,2025-02-17,4.73,4.69,4.69,4.92,5.00,4.12,4.65,1.23,Albany
2,938271786418327576,Knickerbocker 3 The Argus Hotel,Large queen room with queen pull out couch loc...,The Argus Hotel is located directly on Washing...,526358730,Argus,2023-07-17,NaN,NaN,within an hour,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Albany
3,898743867240851780,"Airy 5Br-2Ba,Kids Rm, 6TVs,Pvt Fence-yard&Park...","Whatever brings you to Albany, New York – wish...",NaN,516414095,Rashawn,2023-05-24,"Albany, NY",NaN,within an hour,...,2025-01-26,4.88,4.88,4.86,4.94,4.98,4.80,4.92,2.36,Albany
4,1005978346817864540,Massive 4BD/3BA 2-Fl Townhouse Apt Downtown Al...,Stay in our 4-bedroom townhouse-style 2-story ...,+ 92 Walkscore<br />+ 5-minute walk to MVP Are...,44434571,Earl,2015-09-17,NaN,"I grew up in Cebu, Philippines and moved to Ne...",within an hour,...,2025-01-26,4.80,4.86,4.84,4.86,4.90,4.61,4.76,2.99,Albany


## Dimensiones exactas del dataset

In [21]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 261894 entries, 0 to 261893
Data columns (total 55 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   id                           261894 non-null  int64  
 1   name                         261894 non-null  str    
 2   description                  255801 non-null  str    
 3   neighborhood_overview        127687 non-null  str    
 4   host_id                      261894 non-null  int64  
 5   host_name                    261780 non-null  str    
 6   host_since                   261771 non-null  str    
 7   host_location                195064 non-null  str    
 8   host_about                   135708 non-null  str    
 9   host_response_time           221647 non-null  str    
 10  host_response_rate           221647 non-null  str    
 11  host_acceptance_rate         239348 non-null  str    
 12  host_is_superhost            254521 non-null  str    
 13  host_neigh

## Ejecutamos describe() sobre el **dataset**

In [22]:
df.describe()

,id,host_id,host_listings_count,host_total_listings_count,latitude,longitude,accommodates,bathrooms,bedrooms,beds,...,estimated_occupancy_l365d,estimated_revenue_l365d,review_scores_rating,review_scores_accuracy,review_scores_cleanliness,review_scores_checkin,review_scores_communication,review_scores_location,review_scores_value,reviews_per_month
count,2.618940e+05,2.618940e+05,261771.000000,261771.000000,261894.000000,261894.000000,261894.000000,261769.000000,261542.000000,261404.000000,...,122229.000000,122229.000000,210942.000000,210935.000000,210934.000000,210927.000000,210933.000000,210929.000000,210928.000000,210943.000000
mean,6.124087e+17,2.204047e+08,75.654820,112.045765,32.653782,-13.057419,3.950820,1.393651,1.650263,2.336988,...,84.733803,13755.051755,4.748371,4.777729,4.739261,4.830257,4.838841,4.776649,4.662385,1.435113
std,5.252021e+17,2.031357e+08,413.652997,582.889065,24.486725,71.483957,2.441313,0.790719,1.143739,1.829691,...,92.520170,21124.214462,0.392996,0.377133,0.403333,0.341982,0.343688,0.334551,0.424181,1.725668
min,6.000000e+00,2.900000e+01,0.000000,0.000000,-43.440110,-159.714280,1.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.010000
25%,3.800805e+07,3.525372e+07,1.000000,2.000000,34.126130,-73.959310,2.000000,1.000000,1.000000,1.000000,...,0.000000,0.000000,4.670000,4.730000,4.670000,4.800000,4.820000,4.700000,4.570000,0.300000
50%,7.600672e+17,1.460205e+08,3.000000,4.000000,40.212195,2.160524,4.000000,1.000000,1.000000,2.000000,...,48.000000,5940.000000,4.860000,4.890000,4.860000,4.930000,4.950000,4.880000,4.760000,0.860000
75%,1.109411e+18,4.144555e+08,15.000000,20.000000,45.698861,14.775177,5.000000,2.000000,2.000000,3.000000,...,156.000000,19140.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,4.910000,2.000000
max,1.377118e+18,6.837397e+08,5374.000000,9988.000000,60.025359,153.634660,16.000000,50.000000,50.000000,121.000000,...,255.000000,255000.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,5.000000,82.600000


### Analisis de anomalias

#### Bathrooms(baños)
Me extraño que el minimo de baños, sea 0, osea que no tienen baños. Y que el maximo sea 50, habra algun inmueble con 50 baños?
Comparado con la media que es 1,3 son claramente outliers

#### Beds(camas)
Tambien me parece raro que el minimo de camas sea 0, cuando el objetivo de ir alquiler temporal es por lo menos tener a donde dormir. Y que el maximo sea 121. La media es 2,3 por lo tanto los minimos y maximos son outliers.


# Implementacion de logica python

## Condicionales
En este caso vamos a usar los condicionales para agregar una nueva columna, donde vamos a categorizar de los alojamiento segun su calificacion general.

Para este vamos a usar la columna review_scores_rating.
Vamos a calular su mediana.
Y en base a eso vamos a categorizar:
1. baja
  '< que la media'
2. media
  '= que la media'
3. alta
  '>' que la media





In [23]:
media_review_scores_rating = df['review_scores_rating'].median()
print(f'La media de las review es {media_review_scores_rating}')

df["deseabilidad"] = np.select(
    [
        df['review_scores_rating'] > media_review_scores_rating,
        df['review_scores_rating'] < media_review_scores_rating,
        df['review_scores_rating'] == media_review_scores_rating
    ],
    [
        "alta",
        "baja",
        "media"
    ],
    default="sin_categoria"
)

La media de las review es 4.86


## Funcion lambda para una transformacion atomica
Vamos a aprovechar y vamos a poner todo en mayuscula la columna que creamos para la deseabilidad

In [24]:
df["deseabilidad"] = df["deseabilidad"].apply(lambda x: x.upper())

In [25]:
df.groupby("deseabilidad").size()

deseabilidad
ALTA             103986
BAJA             102340
MEDIA              4616
SIN_CATEGORIA     50952
dtype: int64